llvm/include/llvm/ProfileData/MemProfReader.h

```diff
@@ -97,28 +82,18 @@ class MemProfReader {
   virtual ~MemProfReader() = default;
 
   // Initialize the MemProfReader with the given MemProf profile.
-  MemProfReader(IndexedMemProfData MemProfData) {
-    for (const auto &[FrameId, F] : MemProfData.Frames)
-      IdToFrame.try_emplace(FrameId, F);
-    for (const auto &[CSId, CS] : MemProfData.CallStacks)
-      CSIdToCallStack.try_emplace(CSId, CS);
-    FunctionProfileData = std::move(MemProfData.Records);
-  }
+  MemProfReader(IndexedMemProfData MemProfData)
```

Comment:
Should this be `IndexedMemProfData&&`?

Reply:
Fixed in the latest iteration.  I added `std::move` to a couple of places in `llvm/unittests/ProfileData/MemProfTest.cpp`.